File: backend/app/schemas/system.py
from pydantic import BaseModel


class QueueStatusResponse(BaseModel):
    redis_connected: bool
    celery_available: bool
    pending_task_count: int | None = None
    active_task_count: int | None = None
    failed_recent_count: int | None = None
    api_healthy: bool = True
    postgres_connected: bool | None = None
    qdrant_connected: bool | None = None
    running_index_jobs: int = 0
    pending_index_jobs: int = 0
    chat_today_count: int = 0
    search_today_count: int = 0
    document_upload_today_count: int = 0
    recent_error_count: int = 0
    flower_url: str | None = None
    latest_backup_status: str | None = None
    latest_backup_time: str | None = None
    latest_failed_alert: str | None = None
    tracing_enabled: bool = False
    otlp_endpoint: str | None = None
    apm_enabled: bool = False
    jaeger_url: str | None = None
    loki_enabled: bool = False
    loki_status: str | None = None

File: frontend/src/api/system.ts
import { request } from "./client";

export interface QueueStatus {
  redis_connected: boolean;
  celery_available: boolean;
  pending_task_count: number | null;
  active_task_count: number | null;
  failed_recent_count: number | null;
  api_healthy: boolean;
  postgres_connected: boolean | null;
  qdrant_connected: boolean | null;
  running_index_jobs: number;
  pending_index_jobs: number;
  chat_today_count: number;
  search_today_count: number;
  document_upload_today_count: number;
  recent_error_count: number;
  flower_url?: string | null;
  latest_backup_status?: string | null;
  latest_backup_time?: string | null;
  latest_failed_alert?: string | null;
  tracing_enabled: boolean;
  otlp_endpoint?: string | null;
  apm_enabled: boolean;
  jaeger_url?: string | null;
  loki_enabled: boolean;
  loki_status?: string | null;
}

export interface HealthStatus {
  status: string;
  api: boolean;
  postgres: boolean;
  redis: boolean;
  qdrant: boolean;
  celery: boolean;
  timestamp: string;
}

export const systemApi = {
  queueStatus: () => request<QueueStatus>("/system/queue-status"),
  health: () => request<HealthStatus>("/health"),
};
